pyproject.toml

```diff
@@ -32,39 +32,39 @@ repository = "https://github.com/vprusso/toqito"
 
 [tool.poetry.dependencies]
 python = ">=3.10,<4"
-cvxpy = "1.6.2"
-more-itertools = "10.6.0"
-numpy = "2.2.3"
-scipy = "1.15.2"
-scs = "3.2.7.post2"
-picos = "2.6.0"
+cvxpy = "^1.6.2"
+more-itertools = "^10.6.0"
+numpy = "^2.2.3"
+scipy = "^1.15.2"
+scs = "^3.2.7.post2"
+picos = "^2.6.0"
 
 [tool.poetry.group.dev.dependencies]
-pytest = "8.3.4"
-pytest-cov = "6.0.0"
-coverage = "7.7.0"
-sympy = "1.13.3"
-distlib = "0.3.9"
-ipython = "8.34.0"
+pytest = "^8.3.4"
+pytest-cov = "^6.0.0"
+coverage = "^7.7.0"
+sympy = "^1.13.3"
+distlib = "^0.3.9"
+ipython = "^8.34.0"
 setuptools = ">65.5.1"
 pre-commit = ">=3.0.0"
 
 [tool.poetry.group.lint.dependencies]
-flake8 = "7.1.2"
-flake8-docstrings = "1.7.0"
-isort = "6.0.0"
-myst-parser = "4.0.1"
-pep8 = "1.7.1"
-ruff = "0.11.0"
-mypy = "1.15.0"
+flake8 = "^7.1.2"
+flake8-docstrings = "^1.7.0"
+isort = "^6.0.0"
+myst-parser = "^4.0.1"
+pep8 = "^1.7.1"
+ruff = "^0.11.0"
+mypy = "^1.15.0"
 
 [tool.poetry.group.docs.dependencies]
 readthedocs-sphinx-ext = "*"
-sphinx = "8.1.3"
-furo = "2024.8.6"
-sphinxcontrib-bibtex = "2.6.3"
-sphinx-autoapi = "3.6.0"
-sphinx-copybutton = "0.5.2"
+sphinx = "^8.1.3"
+furo = "^2024.8.6"
+sphinxcontrib-bibtex = "^2.6.3"
+sphinx-autoapi = "^3.6.0"
+sphinx-copybutton = "^0.5.2"
 
 [build-system]
 requires = ["poetry-core>=2.0.0,<3.0.0"]
```

Comment:
I am not too sure of this dependency syntax. We are using `poetry>=2.0` 

https://python-poetry.org/docs/dependency-specification/#projectdependencies-and-toolpoetrydependencies

`^` is used for prior to `2.0`.

Reply:
The `^` symbol is still supported in Poetry 2.0+ according to the same link you provided.

Reply:
Maybe I am not looking at the right section?

![image](https://github.com/user-attachments/assets/2691e9aa-c014-4136-9dc2-1a80e9a98d6e)

I was thinking of switching to `project.dependencies` as we are using `> 2.0`.

Reply:
Yes, but this isn't saying anything about `^` vs `>=`. It's more about `dependencies` vs. `[tool.poetry.dependencies]`. If you ctrl-f for other instance of `^`, you will see that the `^` delimitter is still supported and used in Poetry > 2.0

<img width="856" alt="Screenshot 2025-04-05 at 4 20 09 PM" src="https://github.com/user-attachments/assets/ba1d4dc0-d85e-4055-acf7-f692f16acb93" />

Reply:
The problem is if we decide to switch to `project.dependencies`, we can't use caret syntax. 

![image](https://github.com/user-attachments/assets/ab462496-b63f-4dd2-a335-4d20cbb709f6)

Reply:
@purva-thakre  Oh, this is a good call, and somehow I completely missed that. 

Okay, we definitely don't want to go adding things here that would not be compatible with Poetry 2.0+. 

@ryanbijoy what do you think about using the more modern `project.dependencies` way of specifying packages as described on the official Poetry docs? I think this would be the better alternative so we stay compatible with new versions of Poetry. We also I think want to move to `uv` at some point, so doing this will also allow us to likely make that shift in an easier way.

Reply:
Following up on this thread, @ryanbijoy does using the more modern `project.dependencies` seem like a sensible approach for you to implement? LMK if that seems reasonable to you!